gc.c

```diff
@@ -11748,15 +11748,19 @@ get_envparam_double(const char *name, double *default_value, double lower_bound,
 }
 
 static void
-gc_set_initial_pages(rb_objspace_t *objspace)
+gc_set_initial_pages(rb_objspace_t *objspace, int global_heap_init_slots)
 {
     gc_rest(objspace);
 
     for (int i = 0; i < SIZE_POOL_COUNT; i++) {
         rb_size_pool_t *size_pool = &size_pools[i];
+        char env_key[sizeof("RUBY_GC_HEAP_INIT_SIZE_" "_SLOTS") + DECIMAL_SIZE_OF_BITS(sizeof(size_pool->slot_size) * CHAR_BIT)];
+        snprintf(env_key, sizeof(env_key), "RUBY_GC_HEAP_INIT_SIZE_%d_SLOTS", size_pool->slot_size);
+        size_t pool_init_slots = global_heap_init_slots ? gc_params.heap_init_slots : 0;
+        get_envparam_size(env_key, &pool_init_slots, 0);
```

Comment:
`pool_init_slots` is being overwritten here so the original value is gone.

Reply:
It is only the case that env is set.